app/lib/search/mem_index.dart

```diff
@@ -621,17 +629,42 @@ class _LikeTracker {
     _changed |= removed != null;
   }
 
+  Future<void> _updateScoresIfNeeded() async {
+    if (!_changed) {
+      // we know there is nothing to update
+      return;
+    }
+    final now = DateTime.now();
+    if (_lastUpdated != null && now.difference(_lastUpdated).inHours < 12) {
+      // we don't need to update too frequently
+      return;
+    }
+
+    await _updateScores();
+  }
+
   /// Updates `_LikeScore.score` values, setting them between 0.0 (no likes) to
   /// 1.0 (most likes).
-  void _updateScores() {
+  Future<void> _updateScores() async {
+    final sw = Stopwatch()..start();
     final entries = _values.values.toList();
+
+    // The method could be a single sync block, however, while the index update
+    // happens, we are not serving queries. With the forced async segments,
+    // the waiting queries will be served earlier.
+    await Future.delayed(Duration.zero);
```

Comment:
I think `await null` should do the trick...

Reply:
I always hoped that `await null` will get eventually optimized by the VM. (`Duration.zero` may have the same future...)

Reply:
I'll leave it this way, as other parts of `mem_index.dart` are using the same pattern. We can re-visit in a cleanup.